stdlib/REPL/src/REPL.jl

```diff
@@ -83,7 +99,7 @@ function eval_user_input(@nospecialize(ast), backend::REPLBackend)
                 put!(backend.response_channel, (lasterr,true))
             else
                 backend.in_eval = true
-                value = Core.eval(Main, ast)
+                value = Core.eval(Main, softscope!(ast))
```

Comment:
Maybe it would be better to have an array of transformation functions that are applied to the `ast`, as was discussed in https://github.com/stevengj/SoftGlobalScope.jl/issues/2, where `softscope!` is in the list by default.

This would make future customizations of the REPL easier and more composable, e.g. Revise.jl [hacks the REPL input](https://github.com/timholy/Revise.jl/blob/672dc74b99f2dbd9183abeeed73ff0e65af7781a/src/Revise.jl#L1044-L1054) as well.   It would also allow people to `pop!` the `softscope!` transformation if they don't like it.